.github/workflows/docker-publish.yml

```diff
@@ -35,89 +36,81 @@ jobs:
         with:
           path: |
             ~/.cache/nix
-          key: ${{ runner.os }}-nix-${{ hashFiles('**/docker-image.nix', '**/Cargo.lock') }}
+          key: ${{ runner.os }}-nix-${{ matrix.arch }}-${{ hashFiles('**/docker-image.nix', '**/Cargo.lock') }}
           restore-keys: |
-            ${{ runner.os }}-nix-
+            ${{ runner.os }}-nix-${{ matrix.arch }}-
 
       - name: Build with Nix
         run: |
-          nix-build build-image.nix
-          docker load < result
+          export NIX_BUILD_CORES=$(nproc)
+          nix-build docker-image.nix -A ${{ matrix.arch }} -o result-${{ matrix.arch }}
+          ls -l result-${{ matrix.arch }}
+        env:
+          NIX_BUILD_CORES: ${{ steps.cores.outputs.count }}
+
+      - name: Load image into Docker
+        run: docker load < result-${{ matrix.arch }}
 
-      - name: Verify image contents
+      - name: Verify image contents and entrypoint
         run: |
-          docker run --rm hxckr-core:latest ls -l /app
-          docker run --rm hxckr-core:latest ls -l /app/migrations
-          docker run --rm hxckr-core:latest cat /app/entrypoint.sh
-          docker run --rm hxckr-core:latest diesel --version
+          docker run --rm hxckr-core:${{ matrix.arch }}-latest ls -l /app || echo "Failed to list /app contents"
+          docker run --rm hxckr-core:${{ matrix.arch }}-latest ls -l /app/migrations || echo "Failed to list migrations"
+          docker run --rm hxckr-core:${{ matrix.arch }}-latest cat /app/entrypoint.sh || echo "Failed to cat entrypoint.sh"
+          docker run --rm hxckr-core:${{ matrix.arch }}-latest diesel --version || echo "Failed to get diesel version"
 
       - name: Login to DockerHub
         uses: docker/login-action@v2
         with:
           username: ${{ env.DOCKERHUB_USERNAME }}
           password: ${{ env.DOCKERHUB_TOKEN }}
 
-      - name: Push Docker image
+      - name: Tag and push Docker image
         run: |
-          docker tag hxckr-core:latest ${{ env.IMAGE_NAME }}:dev
-          docker tag hxckr-core:latest ${{ env.IMAGE_NAME }}:${{ github.sha }}
-          docker push ${{ env.IMAGE_NAME }}:dev
-          docker push ${{ env.IMAGE_NAME }}:${{ github.sha }}
+          TAG=${{ github.event_name == 'release' && 'latest' || 'dev' }}
```

Comment:
can the event name be both 'release' and 'latest'?

Reply:
This line uses a conditional expression to determine the value of the `TAG` variable based on the event that triggered the workflow.

   - `${{ github.event_name == 'release' && 'latest' || 'dev' }}`:
    - `github.event_name` is a context variable provided by GitHub Actions that contains the name of the event that triggered the workflow.
     - The expression checks if `github.event_name` is equal to `'release'`.
     - If the event name is `'release'`, the expression evaluates to `'latest'`.
     - If the event name is not `'release'`, the expression evaluates to `'dev'`.

   in shot if the workflow is triggered by a release event, the tag will be `latest`. Otherwise, it will be `dev`.

yea and I forgot we are not usin latest but main for prod. will change that now.  The error was becase  I moved this here from the test push on my fork from friday that we tested.